src/tabs/tab-header-bar.tsx

```diff
@@ -286,7 +354,17 @@ export function TabHeaderBar({
         role="presentation"
         key={tab.id}
       >
-        {trigger}
+        <div className={tabHeaderContainerClasses} role="group" aria-labelledby={commonProps.id}>
+          {trigger}
+          {action && (
+            <span className={styles['tabs-tab-action']} aria-label="More Actions" aria-haspopup="true">
```

Comment:
Why is there an ARIA label and ARIA has-popup? For custom actions we should expect the consumers to set the ARIA label on the button or button-dropdown they provide. Besides, an ARIA label added to a span element w/o interactive role won't be read by screen readers.